library/Fission/Web/IPFS/Pin.hs

```diff
@@ -63,19 +64,10 @@ unpin ::
   -> ServerT UnpinAPI m
 unpin userId cid = do
   remaining <- runDB do
-    delete <| from \userCid ->
-      where_ (selectExact userCid)
+    UserCidMutation.destroyExact userId cid
+    UserCidQuery.getByCids [cid]
 
-    select <| from \userCid -> do
-      where_ (selectExact userCid)
-      limit 1
-      return userCid
-
-  when (null remaining) do
+  when (length remaining == 0) do
```

Comment:
💅 A quick stylistic note: it's generally more common to phrase things declaratively, rather than inspecting a property

```haskell
when (remaining == []) do
  void <| Web.Err.ensure =<< IPFS.Pin.rm cid
```